src/api/pulls.rs

```diff
@@ -369,6 +371,37 @@ impl<'octo> PullRequestHandler<'octo> {
         comment::ListCommentsBuilder::new(self, pr)
     }
 
+    ///creates a new `CommentBuilder` for GET/PATCH/DELETE requests
+    /// to the `/repos/{owner}/{repo}/pulls/{pr}/comments/{comment_id}` endpoint
+    /// ```no_run
+    ///  use octocrab::models::CommentId;
+    /// use octocrab::models::pulls::Comment;
+    ///  async fn run() -> octocrab::Result<Comment> {
+    ///     let octocrab = octocrab::Octocrab::default();
+    ///     let _ = octocrab.pulls("owner", "repo").comment(CommentId(21)).delete();
+    ///     let _ = octocrab.pulls("owner", "repo").comment(CommentId(42)).update("new comment");
+    ///     let comment = octocrab.pulls("owner", "repo").comment(CommentId(42)).get().await;
+    ///
+    ///     comment
+    ///  }
+    /// ```
+    pub fn comment(&self, comment_id: CommentId) -> comment::CommentBuilder {
+        comment::CommentBuilder::new(self, comment_id)
+    }
+
+    /// creates a builder for the `/repos/{owner}/{repo}/pulls/{pull_number}/......` endpoint
+    /// working with particular pull request, e.g.
+    /// * /repos/{owner}/{repo}/pulls/{pull_number}/reviews/{review_id}/events
+    /// * /repos/{owner}/{repo}/pulls/{pull_number}/reviews/{review_id}
+    /// * /repos/{owner}/{repo}/pulls/{pull_number}/commits
+    /// * /repos/{owner}/{repo}/pulls/{pull_number}/reviews/{review_id}/comments
+    /// * /repos/{owner}/{repo}/pulls/{pull_number}/reviews/{review_id}/dismissals
+    /// * /repos/{owner}/{repo}/pulls/{pull_number}/comments/{comment_id}/replies
+    ///
+    pub fn pull_number(&self, pull_nr: u64) -> SpecificPullRequestBuilder {
+        SpecificPullRequestBuilder::new(self, pull_nr)
+    }
```

Comment:
I think instead of having a specific builder for this, we should instead mimic octokit here, and have each of these methods attached to the `pulls` builder, and have the pull number as a required argument.

Reply:
So far it is looking like
```rust
octocrab.pulls("owner", "repo")
    .pull_number(42)
    .reviews()
        .review(42).get()

octocrab.pulls("owner", "repo")
    .pull_number(42)
    .reviews()
        .review(42).update("this is a new body")

octocrab.pulls("owner", "repo")
    .pull_number(42)
    .comment(CommentId(42))
    .reply("new comment")

octocrab.pulls("owner", "repo")
    .pull_number(42)
    .reviews().review(42)
    .list_comments()
        .per_page(10)
        .page(3u32)
        .send()
```
How do you prefer it? Octokit does it like
```c#
[ManualRoute("GET", "/repos/{owner}/{repo}/pulls/{pull_number}/reviews/{review_id}/comments")]
        public Task<IReadOnlyList<PullRequestReviewComment>> GetAllComments(string owner, string name, int number, long reviewId, ApiOptions options)
```
so we would have
-=1=-
```rust
octocrab.pulls("owner", "repo")
        .get_all_comments(/*pr_nr*/ 42, /*review_nr*/ 42)
        .per_page(10)
        .page(3u32)
        .send()
```
or even
-=2=-
```rust
octocrab.pulls("owner", "repo")
        .get_all_comments(/*pr_nr*/ 42, /*review_nr*/ 42, /*pagination*/ PagingOptions{...})
```
?

IMHO, in absence of named arguments, the following options are more readable:
- Builder pattern;
- Struct parameter.

Anyways, I sense we go with -=1=- (to retain Octocrab pagination style)? TIA @XAMPPRocky

Reply:
Yes, 1, I just mean removing the `pull_number` builder and having the pr number as the first argument in these calls.

Reply:
@XAMPPRocky please check if it suits 
https://github.com/XAMPPRocky/octocrab/pull/605/commits/377342b0838f048b2b56b3274075bc25884577fe